### DISCUSSION TITLE
Unknown HTTP status code handling

### DISCUSSION THREAD

**Discussion by rkben** [encode/httpx#2025] [Potential Issue] [answered]
Currently `httpx.RemoteProtocolError` is raised when an unknown status code is returned. Should this be raised at all for status codes > 600?
I assume the common paradigm is to check for a specific status code(s) then fall through otherwise. I would propose this should not raise at all but given a default value of "Unknown" or similar. At the least raise with something a bit more verbose such as `UnknownStatusCode`.

This came up with linkedin, which returns 999 (because of course they do).
```py
import httpx
a = httpx.get("https://www.linkedin.com/in/abc/")
```

**Comment by lovelydinosaur** [comment, marked as answer]
Thanks for noting this @rkben. We're moving to have more relaxed handling here.

This pull request on `h11` resolves the issue - https://github.com/python-hyper/h11/pull/140
Once that's merged and released then the behaviour will be to *not* raise an error on that case.
